roles/kubernetes/control-plane/tasks/kubeadm-secondary.yml

```
@@ -63,6 +63,26 @@
     - kubeadm_already_run is not defined or not kubeadm_already_run.stat.exists
     - not kube_external_ca_mode
 
+- name: Get kubeconfig for join discovery process
+  command: "{{ kubectl }} -n kube-public get cm cluster-info -o jsonpath='{.data.kubeconfig}'"
+  register: kubeconfig_file_discovery
+  run_once: true
+  delegate_to: "{{ groups['kube_control_plane']|first }}"
+  when:
+    - kubeadm_use_file_discovery
+    - kubeadm_already_run is not defined or not kubeadm_already_run.stat.exists
+
+- name: Copy discovery kubeconfig
+  copy:
+    dest: "{{ kube_config_dir }}/discovery-kubeconfig.yaml"
```

Wondering if it should not be a bit more explicit, something like that WDYT?
```suggestion
    dest: "{{ kube_config_dir }}/discovery-cluster-info-kubeconfig.yaml"
```

---

I would go for `cluster-info-discovery-kubeconfig.yaml` instead.